### ISSUE TITLE
Auto import not working bacuase INVALID DATE FOUND

### ISSUE THREAD

**Issue by kekec777** [Jackett/Jackett#4871]
Hallo forum 
im here with a new problem :/
i have a turkish serie that i finaly made to dl automaticly BUT now the parsing is't working, here's  the log:

|Debug|Parser|Parsing string 'Cukur 2.Sezon 23.BLM (11.03.2019) 720p HDTV AC3 x264-TURG.mkv'
|Debug|Parser|Invalid date found: 11/3/2019 12:00:00 AM

So i read about that you could convert the date in .yml file, but it seems i cant make it work..
So please help me or give me direction to how to resolve this / convert this date to something Sonarr can work with (US date)

Thanks in advice :)

**Comment by garfield69** [comment]
I would need to see the html from the site that you are fetching this torrent from, so that we could be sure what the date string actually looks like (in case it contains white spaces, breaks, etc).
But for general date-parsing, this wiki https://github.com/Jackett/Jackett/wiki/Definition-format#dateparse explains the dateparse template you can use to extract the date from the html.
so for input `11/3/2019 12:00:00 AM` the template would be `2\1\2006 03:04:05 PM`

**Comment by kekec777** [comment]
@garfield69 
the site is http://turktorrent.us/
ill try '2/1/2006 03:04:05 PM' and will get back to you after test :)

**Comment by garfield69** [comment]
>This did not work :(

I think you need to provide a better explanation of what you are trying to do.
Download automatically from where, using what app?

show us the code in the indexer that you currently have, and explain what you want to achieve.

**Comment by kekec777** [comment]
> > This did not work :(
> 
> I think you need to provide a better explanation of what you are trying to do.
> Download automatically from where, using what app?
> 
> show us the code in the indexer that you currently have, and explain what you want to achieve.

My goal and procress so far:
I have a turkish serie called the pit / cukur 
In jackett i have 1 indexer that has this serie = turktorrent
SO the first problem WAS that sonarr couldt not find the series or any episodes because the episodes was named like: Cukur 1.sezon 1.blm
So that i got fixed (replace )
Now Sonarr can find the series / episodes and download them , BUT it cant import the file to the correct folder because of this error:
|Debug|Parser|Parsing string 'Cukur 2.Sezon 23.BLM (11.03.2019) 720p HDTV AC3 x264-TURG.mkv'
|Debug|Parser|Invalid date found: 11/3/2019 12:00:00 AM

Here's the yml file:
[turktorrent.ymlTHISONE.log](https://github.com/Jackett/Jackett/files/2964775/turktorrent.ymlTHISONE.log)

**Comment by garfield69** [comment]
OK. then the problem has nothing to do with the date-parsing of the date selector in the fields block.
so put the date selector dateparse filter arg back to `02-01-2006 15:04`, as the other one will generate log errors if you continue using it.

now, if sonarr is choking on (11.03.2019) in the title when its trying to build the folder name, and you think it needs to be a US date, then use your title filters to regex swap the dd.mm into mm.dd 
something like:
```yaml
          - name: re_replace # switch (dd.mm.yyyy) to mm.dd.yyyy
            args: ["\\((\\d+)\\.(\\d+)\\.(\\d+)\\)", "$2/$1/$3"]
```

**Comment by kekec777** [comment]
@garfield69 
That seems to removed the error from log, but now i have another issue where the Season & episodes cant be parsed by Sonarr
Is there anyway to change that aswell?

**Comment by kekec777** [comment]
HERE'S THE LOG:

19-3-19 08:16:11.8|Debug|Parser|Parsing string 'Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv'
19-3-19 08:16:11.8|Debug|Parser|Episode Parsed. Cukur 2 Sezon 24 BLM - 2019-03-18 
19-3-19 08:16:11.8|Debug|Parser|Language parsed: English
19-3-19 08:16:11.8|Debug|QualityParser|Trying to parse quality for Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv
19-3-19 08:16:11.8|Debug|Parser|Quality parsed: HDTV-720p v1
19-3-19 08:16:11.8|Debug|Parser|Release Group parsed: TURG
19-3-19 08:16:11.8|Debug|ParsingService|No matching series Cukur 2 Sezon 24 BLM
19-3-19 08:16:11.8|Debug|Parser|Parsing string 'Cukur S2E24 03/18/2019 720p HDTV AC3 x264-TURG'
19-3-19 08:16:11.8|Debug|Parser|Episode Parsed. Cukur - S02E24 
19-3-19 08:16:11.8|Debug|Parser|Language parsed: English
19-3-19 08:16:11.8|Debug|QualityParser|Trying to parse quality for Cukur S2E24 03/18/2019 720p HDTV AC3 x264-TURG
19-3-19 08:16:11.8|Debug|Parser|Quality parsed: HDTV-720p v1
19-3-19 08:16:11.8|Debug|Parser|Release Group parsed: TURG
19-3-19 08:16:11.8|Debug|Parser|Parsing string 'Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv'
19-3-19 08:16:11.8|Debug|Parser|Episode Parsed. Cukur 2 Sezon 24 BLM - 2019-03-18 
19-3-19 08:16:11.8|Debug|Parser|Language parsed: English
19-3-19 08:16:11.8|Debug|QualityParser|Trying to parse quality for Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv
19-3-19 08:16:11.8|Debug|Parser|Quality parsed: HDTV-720p v1
19-3-19 08:16:11.8|Debug|Parser|Release Group parsed: TURG
19-3-19 08:16:11.8|Debug|ImportDecisionMaker|Analyzing 1/1 files.
19-3-19 08:16:11.8|Debug|Parser|Parsing string 'Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv'
19-3-19 08:16:11.8|Debug|Parser|Episode Parsed. Cukur 2 Sezon 24 BLM - 2019-03-18 
19-3-19 08:16:11.8|Debug|Parser|Language parsed: English
19-3-19 08:16:11.8|Debug|QualityParser|Trying to parse quality for Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv
19-3-19 08:16:11.8|Debug|Parser|Quality parsed: HDTV-720p v1
19-3-19 08:16:11.8|Debug|Parser|Release Group parsed: TURG
19-3-19 08:16:11.8|Debug|VideoFileInfoReader|Getting media info from /downloads/complete/Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv
19-3-19 08:16:11.8|Debug|DetectSample|Runtime is over 90 seconds
19-3-19 08:16:11.8|Debug|Parser|Parsing string 'Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv'
19-3-19 08:16:11.8|Debug|Parser|Episode Parsed. Cukur 2 Sezon 24 BLM - 2019-03-18 
19-3-19 08:16:11.8|Debug|Parser|Language parsed: English
19-3-19 08:16:11.8|Debug|QualityParser|Trying to parse quality for Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv
19-3-19 08:16:11.8|Debug|Parser|Quality parsed: HDTV-720p v1
19-3-19 08:16:11.8|Debug|Parser|Release Group parsed: TURG
19-3-19 08:16:11.8|Debug|VideoFileInfoReader|Getting media info from /downloads/complete/Cukur 2.Sezon 24.BLM (18.03.2019) 720p HDTV AC3 x264-TURG.mkv
19-3-19 08:16:11.8|Warn|ValidateParsedEpisodeInfo|Found daily-style episode for non-daily series: [335574][The Pit]
19-3-19 08:16:11.8|Debug|AggregateQuality|Using quality: HDTV-720p v1
19-3-19 08:16:11.8|Debug|ImportDecisionMaker|File rejected for the following reasons: [Permanent] Invalid season or episode

**Comment by garfield69** [comment]
show me your yaml

**Comment by kekec777** [comment]
[turktorrent.ym.log](https://github.com/Jackett/Jackett/files/2982348/turktorrent.ym.log)

**Comment by garfield69** [comment]
sorry i have no idea what is going on with your sonarr logs.

Using your yaml with my Jackett and Sonarr, I see the two available s2e24 episodes of cukur 
![image](https://user-images.githubusercontent.com/6933240/54626063-4fdec600-4ad5-11e9-9c26-a4bf7ef7c540.png)

and I can manually download the 720p version no problem
![image](https://user-images.githubusercontent.com/6933240/54625598-72241400-4ad4-11e9-9aa3-34bc60c59b4a.png)

but I don't use sonarr regularly so don't have it set up to do automation.

this is a support site for Jackett not Sonarr.

if you have a problem with auto-downloading on sonarr post your problem on the sonarr support site.

BTW you still need to change your dateparse back from '2/1/2006 03:04:05 PM"' to
```yaml
      date:
        selector: td:nth-child(2) div:nth-last-child(1)
        filters:
          - name: dateparse
            args: "02-01-2006 15:04"		
```

**Comment by kekec777** [comment]
This is the answer i get from Sonarr:
Due to the naming Sonarr thinks the series title is Cukur 2 Sezon 24 BLM with the air date of 2019-03-18, which is wrong, but Sonarr doesn’t know that 2 Sezon is season 2 and 24 BLM is episode 24.

Anyway to split or trim the title , season or episode?

**Comment by garfield69** [comment]
when I use Sonarr with the Jackett turktorrent, and go looking for cukur 2x24 I get 2 hits with the english formatted layout.
![image](https://user-images.githubusercontent.com/6933240/54808955-a707b500-4ce6-11e9-832d-113c028291ff.png)
have you upgraded your Jackett to the latest version and are you seeing something different?

**Comment by kekec777** [comment]
@garfield69 
Jackett Version 0.10.886.0

Thats not the issue, the issue is , Sonarr can dl the episode but cannot move the file to the correct serie/season folder because Sonarr doesn’t know that 2 Sezon is season 2 and 24 BLM is episode 24.
Sonarr thinks the series title is Cukur 2 Sezon 24 BLM

![ScreenShot1](https://user-images.githubusercontent.com/20354802/54812853-6bd89800-4c8c-11e9-8b0f-d0dd088e93f9.png)

**Comment by garfield69** [comment]
I don't see what you expect jackett to do about this.
You made jackett rename the search results from turktorrent into English so that sonarr  could use the turktorrent download link to pass the torrent to your download client.
Once that is done that is the end of the involvement from jackett.
The fact that the filenames of the videos within the torrent you downloaded don't match or conform to a naming format that sonarr can recognize as belonging to the series sonarr  is tracking, is not something that has anything to do with jackett :-(
